superagi/controllers/agent.py

```diff
@@ -473,3 +477,45 @@ def get_agent_configuration(agent_id: int,
     db.session.close()
 
     return response
+
+@router.put("/delete/{agent_id}")
+def delete_agent(agent_id: int, Authorize: AuthJWT = Depends(check_auth)):
+    """
+        Delete an existing Agent
+            - Updates the is_deleted flag: Executes a soft delete
+            - AgentExecutions are updated to: "TERMINATED" if agentexecution is created, All the agent executions are updated
+            - AgentExecutionPermission is set to: "REJECTED" if agentexecutionpersmision is created
+            
+        Args:
+            agent_id (int): Identifier of the Agent to delete
+
+        Returns:
+            A dictionary containing a "success" key with the value True to indicate a successful delete.
+
+        Raises:
+            HTTPException (Status Code=404): If the Agent or associated Project is not found or deleted already.
+    """
+    
+    db_agent = db.session.query(Agent).filter(Agent.id == agent_id).first()
+    db_agent_executions = db.session.query(AgentExecution).filter(AgentExecution.agent_id == agent_id).all()
+    db_agent_execution_permission = db.session.query(AgentExecutionPermission).filter(AgentExecutionPermission.agent_id == agent_id).first()
+    
+    
+    if not db_agent or db_agent.is_deleted:
+        raise HTTPException(status_code=404, detail="agent not found")
+    
+    # Deletion Procedure 
+    db_agent.is_deleted = True
+    if db_agent_executions:
+        # Updating all the RUNNING executions to TERMINATED
+        for db_agent_execution in db_agent_executions:
+            db_agent_execution.status = "TERMINATED"
+            
+    if db_agent_execution_permission:
+        db_agent_execution_permission.status = "REJECTED"
+    
+    db.session.commit()
+    
+    return {"success": True}
+
```

Comment:
Can we just handle by HTTP status code we are trying to avoid success/failure messages

Reply:
sure will incorporate this